fix(nestjs): update all dependencies

MeStrak opened the pull request:
This PR updates all NestJS dependencies to NestJS 8. Updated as NestJS 8 has been available since July and the required changes are very minor (just updating some imports).

A very interesting feature which this enables is lazy loading of modules which can reduce startup time and possibly make Konviw suitable for use in a serverless environment: https://docs.nestjs.com/fundamentals/lazy-loading-modules#lazy-loading-modules.

- Updated according to the migration guide: https://docs.nestjs.com/migration-guide
- Builds and runs normally
- Tests are passing successfully
- Updates remove several detected vulnerabilities
  - NPM audit before updates: 60 vulnerabilities (31 moderate, 27 high, 2 critical)
  - NPM audit after updates: 38 vulnerabilities (20 moderate, 16 high, 2 critical)

Commit:
fix(nestjs): update all dependencies

PiiXiieeS commented on package.json:
```diff
@@ -29,27 +29,28 @@
-    "@nestjs/config": "1.0.1",
-    "@nestjs/core": "7.6.18",
-    "@nestjs/platform-express": "7.6.18",
+    "@nestjs/axios": "^0.0.3",
```
Is axios a new dependency with the v8 of nestjs?

PiiXiieeS replied:
@MeStrak
Did you try to run the tests?
I have many typed errors with the data structure returned by Axios. I do not know why those TS errors did not pop up before this upgrade.
Most probably the best would be to address generically the `data` response structure from Confluence with a set of Interfaces.

MeStrak replied:
Hi @PiiXiieeS, that's strange. Yes I ran the tests and they were working fine, apart from the speed tests failing occasionally due to exceeding the time limit by between 10 and 60ms which I think is actually just related to the performance of my laptop.

HTTP Module is extracted to @nestjs/axios in v8. From the migration guide docs:
> HTTP module
The HttpModule exported from the @nestjs/common package has been deprecated and will be removed in the next major release. Instead, please use the @nestjs/axios package (otherwise, there are no API differences).

I just ran it again on the version you merged with the latest updates for body.view and tests are running fine + I can build and run Konviw (and the tests are running ok on the PR pipeline).

Where are you seeing the type errors you mentioned? Did you run npm install to make sure the new deps were installed correctly?

PiiXiieeS replied:
@MeStrak 

Promise<AxiosResponse<any, any>>' is deprecated.ts(6385)
Observable.d.ts(125, 9): The declaration was marked as deprecated here.
(method) Observable<AxiosResponse<any, any>>.toPromise(): Promise<AxiosResponse<any, any>> (+2 overloads)
@deprecated — Replaced with firstValueFrom and lastValueFrom . Will be removed in v8. Details: https://rxjs.dev/deprecations/to-promise

Probably best should be to move to Observables
https://docs.nestjs.com/techniques/http-module

MeStrak replied:
Hmm, rxjs is the only dependency which isn't pinned so I wonder if that could be the reason for this. You might have a different version in your node_modules folder compared to a fresh install or npm ci.

I'll do some testing to see if I can pin that dependency too.

MeStrak replied:
@PiiXiieeS the rxjs dependency is now pinned @7.4.0.

Could you try installing dependencies with npm ci which should clear node_modules ot make sure everything is clean before the install?

PiiXiieeS replied:
Not very clean but with this commit https://github.com/Sanofi-IADC/konviw/pull/288/commits/74b879d6a48ee6b49694f534aea0b82eac083b62 and casting all data responses as `any` I have fixed my type issues.

I have started an interface (commit https://github.com/Sanofi-IADC/konviw/pull/288/commits/7c9d263026416fe3fe7f70b710a998836d3dbba3) for Confluence API which is WIP and we may use in the near future to properly type the responses retrieved from the API.

PiiXiieeS committed:
Merge branch 'main' into fix/nest-8-update

Commit:
chore(dependency maintenance): pin @nestjs/axios and rxjs

Commit:
chore(dependency maintenance): unpin rxjs

Commit:
fix(dependency maintenance): update and pin rxjs to 7.4.0

PiiXiieeS committed:
fix(axios): temporary cast results.data as any

Casting {data} responses with `any` to avoid data types after
the upgrade to next v8.
TODO: improve the typed code, we may create
interfaces to type properly all Confluence reponses.

PiiXiieeS committed:
chore: WIP interface definition for confluence

PiiXiieeS committed:
feat: updated interface for confluence API
